zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -527,6 +535,145 @@ benchmarks! {
+        //* quadratic complexity should not be allowed in substrate blockchains
+        //* assume at first that the last time frame is one block before the current time frame
+        let t = 0;
+        let current_time_frame: TimeFrame = last_offset + t.saturated_into::<u64>();
+        // for current_time_frame in last_offset..=(last_offset + t.saturated_into::<u64>()) {
+        for i in 1..=f {
+            <MarketIdsPerOpenTimeFrame<T>>::try_mutate(current_time_frame, |ids| {
+                // + 31 to not conflict with the markets of MarketIdsPerOpenBlock
+                ids.try_push((i + 31).into())
+            }).unwrap();
+        }
+        // }
+    }: {
+        Pallet::<T>::market_status_manager::<
+            _,
+            MarketIdsPerOpenBlock<T>,
+            MarketIdsPerOpenTimeFrame<T>
+            >(
+                block_number,
+                last_time_frame,
+                current_time_frame,
+                |market_id, market| {
+                    // noop, because weight is already measured somewhere else
+                    Ok(())
+                },
+            )
+            .unwrap();
+    }
+
+    market_resolution_manager {
+        let r in 1..31;
+        let d in 1..31;
+
+        // ensure markets exist
+        for _ in 0..64 {
+            let (_, market_id) = create_market_common::<T>(
+                MarketCreation::Permissionless,
+                MarketType::Categorical(T::MaxCategories::get()),
+                ScoringRule::CPMM
+            )?;
+            // ensure market is reported
+            T::MarketCommons::mutate_market(&market_id, |market| {
+                market.status = MarketStatus::Reported;
+                Ok(())
+            })?;
+        }
+
+        let block_number = 1u64.saturated_into::<T::BlockNumber>();
+
+        let mut r_ids_vec = Vec::new();
+        for i in 1..=r {
+           r_ids_vec.push(i.into());
+        }
+        MarketIdsPerReportBlock::<T>::mutate(block_number, |ids| {
+            *ids = BoundedVec::try_from(r_ids_vec).unwrap();
+        });
+
+        let mut d_ids_vec = Vec::new();
+        for i in 1..=d {
+            // + 31 to not conflict with the markets of MarketIdsPerReportBlock
+            d_ids_vec.push((i + 31).into());
+        }
+        MarketIdsPerDisputeBlock::<T>::mutate(block_number, |ids| {
+            *ids = BoundedVec::try_from(d_ids_vec).unwrap();
+        });
+
+        // TODO(#730) this will get outdated with the merged PR
+        let now: T::BlockNumber = T::DisputePeriod::get() + block_number;
+    }: {
+        Pallet::<T>::resolution_manager(
+            now,
+            |market_id, market| {
+                // noop, because weight is already measured somewhere else
+                Ok(())
+            },
+        ).unwrap();
+    }
+
+    on_initialize_top_overhead {
+        let now = 2u32;
+    }: {
+        if now <= 1u32 {}
```

Comment:
isn't this a dead code as you initialize now = 2 above?

Reply:
The initial thought was to simulate a near exact code as in the `on_initialize` to get an accurate weight representation. The purpose of this `if`, is that the comparison is benchmarked. I know, everybody assumes, that this comparison has a small effect on the weight, but again, I wanted to be as accurate as possible.

Reply:
Fixed with this 3aec79b4a0b530726c1af595056f6c832135ddeb